File: C++/ArrayString/08_maxChunksTomakeSorted.cpp
#include <bits/stdc++.h>
using namespace std;

int maxChunksToSorted(vector<int> &arr)
{
    int n = arr.size();
    if (n == 1)
        return 1;
    if (n == 2)
    {
        if (arr[0] > arr[1])
            return 1;
        else
            return 2;
    }
    int count = 0;
    int i = 0;
    int maxIndex = 0;
    while (i < n)
    {
        maxIndex = max(arr[i], maxIndex);
        if (maxIndex == i)
            count++;
        i++;
    }
    return count;
}

File: Java/ArrayString/08_maxChunksTomakeSorted.java
public int maxChunksToSorted(int[] arr) {
        int count = 0,maxEle = 0, n= arr.length;
        for(int i=0;i<n;i++){
            maxEle = Math.max(maxEle,arr[i]);
            if(maxEle == i) count++;
        }
        return count;
    }